cafehome/views.py

```diff
@@ -1,16 +1,70 @@
-from django.shortcuts import render, HttpResponse
+from django.http import HttpRequest
+from django.shortcuts import render, HttpResponse, redirect
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.views.generic import *
+from .models import Transaction, Computer, Customer
+from .forms import AddComputer
 
-# Create your views here.
 
-# def home(request):
-#     return render(request, "base.html")
+class Home(TemplateView, LoginRequiredMixin):
+    template_name = "html/cafeehtml/dashboard.html"
 
-class Home(TemplateView):
-    template_name = "home.html"
-    
-    def get_context_data(self, **kwargs):
-        context = {"name":"Mohit"}
+    def get_context_data(self):
+        if self.request.user.is_authenticated:
+            transactions = Transaction.objects.filter(customer_id=self.request.user.id)
```

Comment:
if you have applied related name intransaction models then only you need to write user.transactions.all()

Reply:
permission based